From 16a7eefd1ad815da7c5d96ea6a3b7774f78e96b4 Mon Sep 17 00:00:00 2001
From: Don Brace <don.brace@microsemi.com>
Date: Thu, 20 Dec 2018 19:32:12 -0500
Subject: [PATCH 6/6] scsi: smartpqi: fix build warnings

Fix kbuild warning and fallout from linux-next -Wimplicit-fallthrough.

[mkp: added fall through statements]

Signed-off-by: Don Brace <don.brace@microsemi.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
---
 drivers/scsi/smartpqi/smartpqi_init.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/scsi/smartpqi/smartpqi_init.c b/drivers/scsi/smartpqi/smartpqi_init.c
index 56344cc820f6..7c67285adb99 100644
--- a/drivers/scsi/smartpqi/smartpqi_init.c
+++ b/drivers/scsi/smartpqi/smartpqi_init.c
@@ -476,6 +476,7 @@ static int pqi_build_raid_path_request(struct pqi_ctrl_info *ctrl_info,
 		break;
 	case BMIC_SENSE_DIAG_OPTIONS:
 		cdb_length = 0;
+		/* fall through */
 	case BMIC_IDENTIFY_CONTROLLER:
 	case BMIC_IDENTIFY_PHYSICAL_DEVICE:
 		request->data_direction = SOP_READ_FLAG;
@@ -485,6 +486,7 @@ static int pqi_build_raid_path_request(struct pqi_ctrl_info *ctrl_info,
 		break;
 	case BMIC_SET_DIAG_OPTIONS:
 		cdb_length = 0;
+		/* fall through */
 	case BMIC_WRITE_HOST_WELLNESS:
 		request->data_direction = SOP_WRITE_FLAG;
 		cdb[0] = BMIC_WRITE;
@@ -7469,7 +7471,7 @@ static int pqi_ofa_alloc_mem(struct pqi_ctrl_info *ctrl_info,
 	dev = &ctrl_info->pci_dev->dev;
 
 	sg_count = (total_size + chunk_size - 1);
-	do_div(sg_count, chunk_size);
+	sg_count /= chunk_size;
 
 	ofap = ctrl_info->pqi_ofa_mem_virt_addr;
 
-- 
2.20.1

